### PR TITLE
Added PixiParser

### PR DESCRIPTION
**Pull request by CPKreuz**
Implemented new [PixiParser](https://github.com/PixiEditor/PixiParser)

### CODE REVIEW

**Comment by flabbet** [comment]
Fix tests

**Comment by flabbet** [comment]
Found a bug, create a new file, size 1x1, and save. Then try to open it.
image

**Comment by CPKreuz** [comment]
Fixed in the latest commit, you just have make a NuGet package
The build in the pipeline is gonna fail

**Review by flabbet — CHANGES_REQUESTED** [review]
Updated package, please fix tests

**Review by flabbet — CHANGES_REQUESTED** [review]
Now saving a file and opening it results in the same error.

**Comment by flabbet** [comment]
The first impression is good, I'll test it more a bit later

File: PixiEditor/Helpers/Extensions/ParserHelpers.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        public static Parser.SerializableDocument ToSerializable(this Document document)
+        {
+            Parser.SerializableDocument serializable = new Parser.SerializableDocument();
+
+            serializable.Width = document.Width;
+            serializable.Height = document.Height;
+            serializable.Layers = document.Layers.Select(x => x.ToSerializable()).ToArray();
+            serializable.Swatches = document.Swatches.Select(x => new Tuple<byte, byte, byte, byte>(x.A, x.R, x.G, x.B)).ToArray();
+
+            return serializable;
```
Comment (flabbet) [inline PixiEditor/Helpers/Extensions/ParserHelpers.cs:54]: You can make it shorter.

```suggestion
            Parser.SerializableDocument serializable = new Parser.SerializableDocument()
           {
               Width = document.Width;
               Height = document.Height;
               Layers = document.Layers.Select(x => x.ToSerializable()).ToArray();
               Swatches = document.Swatches.Select(x => new Tuple<byte, byte, byte, byte>(x.A, x.R, x.G, x.B)).ToArray();
            }
            return serializable;
```

File: PixiEditor/Helpers/Extensions/ParserHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            serializable.Height = layer.Height;
+            serializable.BitmapBytes = layer.ConvertBitmapToBytes();
+            serializable.IsVisible = layer.IsVisible;
+            serializable.OffsetX = (int)layer.Offset.Left;
+            serializable.OffsetY = (int)layer.Offset.Top;
+            serializable.Opacity = layer.Opacity;
+            serializable.MaxWidth = layer.MaxWidth;
+            serializable.MaxHeight = layer.MaxHeight;
+
+            return serializable;
```
Comment (flabbet) [inline PixiEditor/Helpers/Extensions/ParserHelpers.cs:72]: Same here

**Review by flabbet — CHANGES_REQUESTED** [review]
It works great, I didn't find any bugs, just small formatting changes :)

**Review by flabbet — APPROVED** [review]
👌

**Pull request merged**